controllers/spacebindingrequest/spacebindingrequest_controller.go

```diff
@@ -161,6 +161,32 @@ func (r *Reconciler) getSpaceBinding(ctx context.Context, spaceBindingRequest *t
 		return nil, nil
 	}
 
+	// more than one spacebinding
+	if len(spaceBindings.Items) > 1 {
+		return nil, fmt.Errorf("expected 1 spacebinding for Space %s and MUR %s. But found %d", space.GetName(), spaceBindingRequest.Spec.MasterUserRecord, len(spaceBindings.Items))
+	}
+
+	// check if existing spacebinding was created from spaceBindingRequest
+	if len(spaceBindings.Items) == 1 {
+		// check if it has the expected spaceBindingRequest labels
+		sbrLabel, sbrLabelFound := spaceBindings.Items[0].Labels[toolchainv1alpha1.SpaceBindingRequestLabelKey]
+		if !sbrLabelFound || sbrLabel == "" {
+			return nil, fmt.Errorf("A SpaceBinding for Space %s and MUR %s already exists. But it doesn't have the expected SpaceBindingRequest label set: %s", space.GetName(), spaceBindingRequest.Spec.MasterUserRecord, toolchainv1alpha1.SpaceBindingRequestLabelKey)
+		}
+		sbrNamespaceLabel, sbrNamespaceLabelFound := spaceBindings.Items[0].Labels[toolchainv1alpha1.SpaceBindingRequestNamespaceLabelKey]
+		if !sbrNamespaceLabelFound || sbrNamespaceLabel == "" {
+			return nil, fmt.Errorf("A SpaceBinding for Space %s and MUR %s already exists. But it doesn't have the expected SpaceBindingRequestNamespace label set: %s", space.GetName(), spaceBindingRequest.Spec.MasterUserRecord, toolchainv1alpha1.SpaceBindingRequestNamespaceLabelKey)
+		}
+		// check if spaceBinding has correct reference to spaceBindingRequest
+		// this should never happen, but you never know ¯\_(ツ)_/¯
+		if sbrLabel != spaceBindingRequest.GetName() {
+			return nil, fmt.Errorf("expected SpaceBindingRequest label: %s, got: %s", spaceBindingRequest.GetName(), sbrLabel)
+		}
+		if sbrNamespaceLabel != spaceBindingRequest.GetNamespace() {
+			return nil, fmt.Errorf("expected SpaceBindingRequestNamespace label: %s, got: %s", spaceBindingRequest.GetNamespace(), sbrNamespaceLabel)
+		}
```

Comment:
We don't need to deal with each of the situation separately, so we could simplify this part into:
```suggestion
		sbrLabel := spaceBindings.Items[0].Labels[toolchainv1alpha1.SpaceBindingRequestLabelKey]
		sbrNamespaceLabel := spaceBindings.Items[0].Labels[toolchainv1alpha1.SpaceBindingRequestNamespaceLabelKey]
		if sbrLabel != spaceBindingRequest.GetName() || sbrNamespaceLabel != spaceBindingRequest.GetNamespace() {
			return nil, fmt.Errorf("A SpaceBinding for Space '%s' and MUR '%s' already exists, but it's not managed by this SpaceBindingRequest CR. It's not allowed to create multiple SpaceBindings for the same combination of Space and MasterUserRecord", space.GetName(), spaceBindingRequest.Spec.MasterUserRecord)
		}
```

Reply:
Thanks for you suggestion. I agree , this will make the code shorter and the error message more "end user friendly". But I think it will make a bit more difficult to understand what actually went wrong, is there some mismatch/misconfiguration with labels or is there actually because of another spacebinding that is not managed by the SpaceBindingRequest. But we can find that out by investigating a bit the SpaceBinding and SBR objects in the cluster eventually.